lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -819,6 +819,100 @@ spec = do
+        let destination = (addrs !! 1) ^. #id
+        let (feeMin, _) = ctx ^. #_feeEstimator $ PaymentDescription
+                { nInputs = 1
+                , nOutputs = 1
+                , nChanges = 1
+                }
+
+        let amt = walletBalance - feeMin :: Natural
+        let payload1 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+
+        r1 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload1
+        verify r1
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
+
+        let payload2 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt+1},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+        r2 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload2
+        verify r2
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
+
+        let (feeMin', _) = ctx ^. #_feeEstimator $ PaymentDescription
+                { nInputs = 1
+                , nOutputs = 1
+                , nChanges = 0
+                }
+        let amt1 = walletBalance - feeMin' :: Natural
+
+        let payload3 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt1},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+        r3 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload3
+        verify r3
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
+
+        let payload4 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt1+1},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+
+        r4 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload4
+        verify r4
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
```

Comment:
I think this should really be captured as a unit test, integration level are slow and rather unnecessary here since this isn't an integration issue or something for which we need to observe the chain to assert about.

Reply:
yeah, unit testing of shelley is better as shown https://github.com/input-output-hk/cardano-wallet/pull/1782#pullrequestreview-435072018